### ISSUE TITLE
The need of detach() on self.netF(self.var_H).detach()

### ISSUE THREAD

**Issue by huuquan1994** [xinntao/BasicSR#50]
Hi,

I've been reading your code and wonder if the `detach` function on NetF is necessary.
In [`SRRaGAN_model.py`](https://github.com/xinntao/BasicSR/blob/master/codes/models/SRRaGAN_model.py), line 128: `real_fea = self.netF(self.var_H).detach()`, since we don't need to update the NetF, is that OK to remove .detach() function?
If no, then why the `fake_fea = self.netF(self.fake_H)` doesn't have the detach function?

____________________________________________
[Update]
Sorry but I think I understood your code. This line is for updating the NetG then we don't need to detach the `self.netF(self.fake_H)` (because `self.fake_H` is the result of NetG). Instead, we do `self.netF(self.var_H).detach()` since it is not necessary.
Please close this topic if my understanding is correct.

**Comment by xinntao** [comment]
Hi @huuquan1994,
Your understanding is right. 
`self.netF(self.var_H).detach()`, we do not need to update NetF, so we detach it. It will stop the calcalation of backpropagation to NetF and thus save the computation time. Otherwise the gradients will further back-propatation to the NetF and they are uselee, wasting the computation.
